File: src/main/java/com/durhack/sharpshot/nodes/math/MultNode.java
package com.durhack.sharpshot.nodes.math;

import com.durhack.sharpshot.gui.Triangle;
import javafx.scene.Node;
import javafx.scene.paint.Color;
import org.jetbrains.annotations.NotNull;

import java.math.BigInteger;

public class MultNode extends MathNode {

    @Override
    public BigInteger operation(BigInteger val1, BigInteger val2) {
        return val1.multiply(val2);
    }

    @Override
    public String toString() {
        return "Multiply";
    }

    @Override
    public @NotNull Node toGraphic() {
        return new Triangle(getRotation(), Color.web("#8df8e6"), "x");
    }
}

File: src/main/java/com/durhack/sharpshot/nodes/math/MultNode.kt
package com.durhack.sharpshot.nodes.math

import com.durhack.sharpshot.gui.shapes.Triangle
import javafx.scene.paint.Color
import java.math.BigInteger

class MultNode : AbstractMathNode() {
    public override fun operation(val1: BigInteger, val2: BigInteger): BigInteger? = val1.multiply(val2)
    override val type = "multiply"
    override fun graphic() = Triangle(rotation, Color.web("#add8e6"), "x")
    override val tooltip = "Multiplies two numbers"
}
